File: javascript/linkedList.js
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
    this.tail = null;
  }

  push(data) {
    const newNode = new Node(data);

    if (!this.head) {
      this.head = newNode;
      this.tail = newNode;
    } else {
      this.tail.next = newNode;
      this.tail = newNode;
    }
  }

  pop() {
    if (this.head == null) {
      console.log("Linked list underflow");
    }
    let poppedValue = this.head;
    this.head = this.head.next;
    return poppedValue;
  }

  print() {
    let current = this.head;
    let output = "";
    while (current !== null) {
      output += current.data + " -> ";
      current = current.next;
    }
    output += "null";
    console.log(output);
  }
}

const list = new LinkedList();

list.push(19);
list.push(23);
list.push(34);
list.print();
list.pop();
list.print();

File: python/linkedList.py
from typing import Optional


class Node:
    def __init__(self, data: int):
        self.data: int = data
        self.next: Optional['Node'] = None


class LinkedList:
    def __init__(self):
        # head and tail will be Node and None
        self.head: Optional['Node'] = None
        self.tail: Optional['Node'] = None

    def inputValue(self) -> int:
        return int(input("Enter value: "))

    def push(self) -> None:
        newNode = Node(self.inputValue())
        if self.tail is None:
            self.head = newNode
            self.tail = newNode
        else:
            self.tail.next = newNode
            self.tail = newNode

    def pop(self) -> Optional[Node]:
        if self.head is None:
            print("linked list underflow")
            return None
        value = self.head
        self.head = self.head.next
        return value

    def print(self):
        current = self.head
        while (current != None):
            print(current.data, " -> ", end=" ")
            current = current.next
        print(" ")
        if self.head is None:
            print("linked list underflow")

    def menu(self) -> None:
        print("1. push")
        print("2. pop")
        print("3. print")
        print("4. exit")


if __name__ == "__main__":
    ll: LinkedList = LinkedList()
    while (True):
        ll.menu()
        choice = int(input("Choice: "))
        if choice == 1:
            ll.push()
        elif choice == 2:
            ll.pop()
        elif choice == 3:
            ll.print()
        elif choice == 4:
            print("Thanks for using linked list")
            break
